### ISSUE TITLE
Using attribute with RelatedList doesn't work

### ISSUE THREAD

**Issue by AbdealiLoKo** [marshmallow-code/marshmallow-sqlalchemy#340]
When I am using a RelatedList with an `attribute=` loading the relationship seems to cause an error about unknown attributes.

For example, if the Models are:
```python
class School(Base):
    __tablename__ = "school"
    id = sa.Column("school_id", sa.Integer, primary_key=True)
    name = sa.Column(sa.String(255), nullable=False)

class Teacher(Base):
    __tablename__ = "teacher"
    id = sa.Column(sa.Integer, primary_key=True)

    current_school_id = sa.Column(
        sa.Integer, sa.ForeignKey(School.id), nullable=True
    )
    current_school = relationship(School, backref=backref("teachers"))
```

My schema is defined as:
```python
class SchoolSchema(SQLAlchemySchema):
    class Meta:
        model = models.School
        sqla_session = session
        load_instance = True

    all_teachers = RelatedList(Related(), attribute="teachers")
```

Now when I try to use is:
```python
schema = SchoolSchema()
out_school = schema.load({"all_teachers": [1]})
```

I get the error:
```
AttributeError: type object 'School' has no attribute 'all_teachers'
```

**Comment by AbdealiLoKo** [comment]
After creating the PR - I saw that there is already a test `test_model_schema_with_attribute`
https://github.com/marshmallow-code/marshmallow-sqlalchemy/blob/7cdb69f136c4de64afae5fefff9c9d47c9d32ad4/tests/test_model_schema.py#L393-L404

where it shows that I would need to do:
```
all_teachers = RelatedList(Related(attribute="teachers"), attribute="teachers")
```
But I'm not sure if that is intuitive, because the attribute should be for the main field in the schema ?
Ideally the RelatedList should pass that onto the inner child during automatically ?

I see that this isn't a problem when using `field_for()` or `auto_field()` because it adds the same set of kwargs to both the inner field and RelatedField:
https://github.com/marshmallow-code/marshmallow-sqlalchemy/blob/7cdb69f136c4de64afae5fefff9c9d47c9d32ad4/src/marshmallow_sqlalchemy/convert.py#L141-L149

**Comment by AbdealiLoKo** [comment]
Seems like I have another issue with RelatedList and `field_for()` with RelatedList.
If I add a custom validator to a field like:
```python
field_for(School, 'teachers', validate=lambda x: len(set(x)) == len(x))
```
It throws an error that Teacher is not itearable. Becuase this validator is also being applied to both: `RelatedList` and `Related`.

@sloria - Wanted to check if it makes sense and whether a fix to decouple the kwargs for RelatedList and Related would be accepted ?
I can attempt to create a PR

**Comment by sloria** [comment]
@AbdealiJK Yeah, that does look like a a bug--`validate` should apply to `RelatedList` and not `Related`. A PR would be great!

**Comment by sloria** [comment]
I think the OP is actually not a bug. i believe what you want is `data_key`, not `attribute`.

See https://marshmallow.readthedocs.io/en/3.x-line/marshmallow.fields.html

> data_key ([str](https://docs.python.org/3/library/stdtypes.html#str) | None) – The name of the dict key in the external representation, i.e. the input of load and the output of dump. If [None](https://docs.python.org/3/library/constants.html#None), the key will match the name of the field.

So i think what you want is

```python
class SchoolSchema(SQLAlchemySchema):
    class Meta:
        model = models.School
        sqla_session = session
        load_instance = True

    teachers = RelatedList(Related(), data_key="all_teachers")
```

closing this for now. i've opened up a separate issue for the validator bug
